### DISCUSSION TITLE
How to calculate the CTC Loss by jax and flax?

### DISCUSSION THREAD

**Discussion by Du-Yao** [google/flax#843] [Q&A]
I want to do the Sign Language Recognition task and Speech Recognition task by jax, and the CTC Loss is the one of the most important loss function. In Tensorflow, there are tf.nn.ctc_loos and tf.keras.backend.ctc_batch_cost APIs.
So how can I use it in jax?

**Comment by jheek** [comment]
Currently neither Jax or Flax maintain any loss functions. CTC is definitely complex compared to an L2 or cross entropy loss.
So it would be nice to have a standard implementation either in Jax/Flax.

There is an internal implementation of CTC that we can hopefully open-source soon.
